In [1]:
import numpy as np

Ecuación que se va a usar:
* $n = \frac{\sin\theta_1}{\sin\theta_2}$

Error de la ecuación:
* $\delta n = \frac{\cos\theta_1}{\sin\theta_2}\delta\theta_1+\frac{\sin\theta_1\cot\theta_2}{\sin\theta_2}\delta\theta_2$

In [7]:
#medidas de agua base
thetaA1 = 48.5*(np.pi/180)
thetaA2 = 41.3*(np.pi/180)
#medidas de agua con 5g de azucar
thetaB1 = 46.4*(np.pi/180)
thetaB2 = 39.7*(np.pi/180)
#medidas de agua con 10g de azucar
thetaC1 = 46.7*(np.pi/180)
thetaC2 = 39.2*(np.pi/180)
#medidas de agua con 15g de azucar
thetaD1 = 49.1*(np.pi/180)
thetaD2 = 40.7*(np.pi/180)
#medidas de agua con 20g de azucar
thetaE1 = 47.7*(np.pi/180)
thetaE2 = 41.5*(np.pi/180)
#medida de la gaseosa que dice tener 18g de azucar
thetaF1 = 37.1*(np.pi/180)
thetaF2 = 21.3*(np.pi/180)
#Error
thetaR = 10*(np.pi/180)

In [3]:
def n(theta1,theta2):
  return (np.sin(theta1))/(np.sin(theta2))

def nr(theta1,theta2,thetaR):
  return ((np.cos(theta1)*thetaR)/(np.sin(theta2))) + ((np.sin(theta1)*(1/np.tan(theta2)*thetaR))/np.sin(theta2))

In [9]:
#indices de refracción
nA = n(thetaA1,thetaA2)
nB = n(thetaB1,thetaB2)
nC = n(thetaC1,thetaC2)
nD = n(thetaD1,thetaD2)
nE = n(thetaE1,thetaE2)
nF = n(thetaF1,thetaF2)

In [10]:
print("Agua base:",nA)
print("Agua 5g:",nB)
print("Agua 10g:",nC)
print("Agua 15g:",nD)
print("Agua 20g:",nE)
print("Gaseosa:",nF)

Agua base: 1.134778527307192
Agua 5g: 1.1337012316212758
Agua 10g: 1.151485784897429
Agua 15g: 1.1591095220104748
Agua 20g: 1.1162220294575191
Gaseosa: 1.6605807142342333


In [11]:
#error de indice de refracción
nrA = nr(thetaA1,thetaA2,thetaR)
nrB = nr(thetaB1,thetaB2,thetaR)
nrC = nr(thetaC1,thetaC2,thetaR)
nrD = nr(thetaD1,thetaD2,thetaR)
nrE = nr(thetaE1,thetaE2,thetaR)
nrF = nr(thetaF1,thetaF2,thetaR)

In [12]:
print("Agua base:",nrA)
print("Agua 5g:",nrB)
print("Agua 10g:",nrC)
print("Agua 15g:",nrD)
print("Agua 20g:",nrE)
print("Gaseosa:",nrF)

Agua base: 0.40066799754338994
Agua 5g: 0.42676072251674535
Agua 10g: 0.4358028452733653
Agua 15g: 0.4104388360871851
Agua 20g: 0.3974714038714112
Gaseosa: 1.1265839521413414


| | Indice de refracción | Error |
|---|---|---|
| Agua base | 1.13 | 0.4 |
| Agua 5g | 1.13 | 0.4 | 
| Agua 10g | 1.15 | 0.4 | 
| Agua 15g | 1.16 | 0.4 | 
| Agua 20g | 1.12 | 0.4 | 
| Gaseosa (18g) | 1.7 | 1.12 | 